File: src/main/java/com/smarthome/backend/model/WaitConfig.java
package com.smarthome.backend.model;

import com.google.gson.annotations.SerializedName;

import lombok.Data;

/**
 * Konfiguration für einen Warte-Knoten im Workflow.
 */
@Data
public class WaitConfig {
    private String type; // 'time' | 'trigger'
    
    @SerializedName("waitTime")
    private Integer waitTime; // Optional, Anzahl der Sekunden zum Warten (für 'time' Typ)
    
    @SerializedName("deviceId")
    private String deviceId; // Optional, ID des Geräts für 'trigger' Typ
    
    @SerializedName("triggerEvent")
    private String triggerEvent; // Optional, Name des Trigger-Events für 'trigger' Typ
    
    @SerializedName("triggerValues")
    private java.util.List<Object> triggerValues; // Optional, Werte für den Trigger

    @SerializedName("timeout")
    private Integer timeout; // Optional, Timeout in Sekunden
}

File: src/main/node/com/smarthome/backend/model/WaitConfig.ts
/**
 * Konfiguration für Warte-Knoten im Workflow.
 * Kompatibel mit Frontend-Datenmodell.
 * 
 * @property type - Typ des Wartens: 'time' | 'trigger'
 * @property waitTime - Wartezeit in Millisekunden (für type='time')
 * @property deviceId - ID des Geräts für Trigger-Event (für type='trigger')
 * @property moduleId - ID des Moduls des Geräts
 * @property triggerEvent - Name des Trigger-Events
 * @property triggerValues - Parameterwerte für den Trigger
 * @property timeout - Timeout in Millisekunden (für type='trigger')
 */
export class WaitConfig {
  type?: string; // 'time' | 'trigger'
  waitTime?: number;
  deviceId?: string;
  moduleId?: string;
  triggerEvent?: string;
  triggerValues?: unknown[];
  timeout?: number;

  constructor(init?: Partial<WaitConfig>) {
    Object.assign(this, init);
  }
}
